blocks/bricks/sequence_generators.py

```diff
@@ -436,6 +436,9 @@ def emit(self, readouts):
 
     @application
     def cost(self, readouts, outputs):
+        # WARNING: unfortunately this application method works
+        # just fine when `readouts` and `outputs` have
+        # different dimensions. Be careful!
         probs = self.probs(readouts)
```

Comment:
There's the `Assert` op, but reading the documentation I'm not sure how user-friendly it is: https://github.com/Theano/Theano/blob/master/theano/tensor/opt.py#L1486-L1495

If you would want to integrate this into Blocks, I guess the way to go would be: Use the `assert_` function in a Brick, add a useful message and tag the output variable with an `ASSERTION` role. We could then have an extension that basically compiles a function that returns all the `ASSERTION` variables, and runs a single batch of a data stream before training starts. If one of the assertions fails, it would give a useful error before training starts, otherwise it could just claim success and start training.